src/blocks/plugins/feedback/index.tsx

```diff
@@ -0,0 +1,185 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	useEffect,
+	useState
+} from '@wordpress/element';
+import {
+	Button,
+	Modal,
+	Spinner,
+	TextareaControl
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { addFilter } from '@wordpress/hooks';
+
+/**
+ * Internal Dependencies
+ */
+import './editor.scss';
+import ButtonVariant = Button.ButtonVariant;
+
+// @ts-ignore
+const { otterVersion, siteURL, assetsPath } = window.themeisleGutenberg ? window.themeisleGutenberg : window.otterObj;
+const finishIcon = assetsPath + ( '/' === assetsPath[ assetsPath.length - 1 ] ? '' : '/' ) + 'icons/finish-feedback.svg';
+
+const collectedInfo = [
+	{
+		name: __( 'Plugin version',  'otter-blocks' ),
+		value: otterVersion
+	},
+	{
+		name: __( 'Current website', 'otter-blocks' ),
+		value: siteURL
+	},
+	{
+		name: __( 'Uninstall reason', 'otter-blocks' ),
+		value: __( 'Text from the above text area', 'otter-blocks' )
+	}
+];
+
+/**
+ * Displays a button that opens a modal for sending feedback
+ *
+ * @param source The area from where the feedback is given
+ * @param text Text to display inside the button
+ * @param variant Variant of the button
+ * @returns {JSX.Element}
+ */
+const Feedback = (
+	source,
+	text = __( 'Help us improve', 'otter-blocks' ),
+	variant:ButtonVariant = 'link'
+) => {
+	const [ isOpen, setIsOpen ] = useState( false );
+	const [ feedback, setFeedback ] = useState( '' );
+	const [ status, setStatus ] = useState( 'notSubmitted' );
+	const [ showInfo, setShowInfo ] = useState( false );
+
+	useEffect( () => {
+		const info = document.querySelector( '.o-feedback-modal .info' ) as HTMLElement;
+		if ( ! info ) {
+			return;
+		}
+
+		info.style.height = showInfo ? `${ info.querySelector( '.wrapper' ).clientHeight }px` : '0';
+	}, [ showInfo ]);
+
+	const sendFeedback = () => {
+		setStatus( 'loading' );
+
+		fetch( 'https://api.themeisle.com/tracking/feedback', {
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/json'
+			},
+			body: JSON.stringify({
+				slug: 'otter-blocks',
+				version: otterVersion,
+				feedback,
+				data: {
+					site: siteURL,
+					source
+				}
+			})
+		}).then( r => {
+			setStatus( 'submitted' );
+		});
+	};
```

Comment:
It will be nice to have an error handler. If an error is happening and nothing shows, the user might think that the feedback was sent. Thus we lose some information.